client/src/components/Browse/Sources/BrowseSources.tsx

```diff
@@ -22,107 +31,128 @@ export const BrowseSources = () => {
   let interactionSources = interactionData?.sources?.nodes;
   let potentiallyDruggableSources = potentiallyDruggableData?.sources?.nodes;
 
-  useEffect(() => {
-    console.log('geneData', geneData)
-  }, [geneData])
+  const sectionsMap = [
+    {
+      heading: "Gene Sources",
+      sources: geneSources,
+      value: "Gene",
+    },
+    {
+      heading: "Drug Sources",
+      sources: drugSources,
+      value: "Drug",
+    },
+    {
+      heading: "Interaction Sources",
+      sources: interactionSources,
+      value: "Interaction",
+    },
+    {
+      heading: "Potentially Druggable Sources",
+      sources: potentiallyDruggableSources,
+      value: "Potentially Druggable",
+    },
+  ]
+
+  const handleButtonClick = (event: any) => {
+    const value = event.target.value
+    setFilter(value)
+  }
 
-  useEffect(() => {
-    console.log('drugData', drugData)
-  }, [drugData])
+  const getCard = (src: any) => {
```

Comment:
Same issue here with the "any" type. This one should play nicer than the above event type since this should just be an object. Is the correct practice to define this entire object as the type here? It'll be pretty lengthy

Reply:
As you point out this one is a distinct problem from the event handler, it can be typed. But as you also note, defining the entire type inline here would be cumbersome and annoying. 

This goes back to the link in one of the issues about `any` types where you can generate type definitions for your graphql queries. That way the data you get back in the very beginning from the various`useSources` hooks is already typed (instead of the `any` you currently get back).

At that point you shouldn't really even have to specify the types manually at all, the compiler will infer them and that will propagate throughout the component.